core/src/main/java/org/apache/calcite/rel/core/Sample.java

```diff
@@ -36,12 +36,12 @@
 public class Sample extends SingleRel {
   //~ Instance fields --------------------------------------------------------
 
-  private final RelOptSamplingParameters params;
+  private final RelOptSamplingRowLimitParameters params;
```

Comment:
This wasn't what I meant. The primary purpose of creating a new class for the parameters was to separate the fixed row version of the sampling by percentage version. This just moves it to a new name but doesn't fix the underlying problem where we're adding extra complexity.

I think my preferred version of this would be to not touch `core/Sample.java` at all. I think we should use it as a base and create a new `RowSample` operation with only the parameters for row sampling.

The only changes to the core packages that should happen is `SqlToRelConverter` and that's only because we haven't refactored that out yet.